src/d3a/d3a_core/sim_results/kpi.py

```diff
@@ -0,0 +1,145 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from d3a.models.strategy.infinite_bus import InfiniteBusStrategy
+from d3a.d3a_core.sim_results.area_statistics import _is_load_node, _is_prosumer_node
+from d3a.models.strategy.pv import PVStrategy
+from d3a.models.strategy.predefined_pv import PVPredefinedStrategy, PVUserProfileStrategy
+from d3a.models.strategy.commercial_producer import CommercialStrategy
+
+
+class KPI:
+    def __init__(self):
+        self.performance_indices = dict()
+        self.producer_list = list()
+        self.consumer_list = list()
+        self.consumer_area_list = list()
+        self.ess_list = list()
+        self.buffer_list = list()
+        self.total_energy_demanded_wh = 0
+        self.total_energy_produced_wh = 0
+        self.total_self_consumption_wh = 0
+        self.self_consumption_buffer_wh = 0
+
+    def __repr__(self):
+        return f"KPI: {self.performance_indices}"
+
+    def _accumulate_devices(self, area):
+        for child in area.children:
+            if type(child.strategy) in [PVStrategy, PVUserProfileStrategy,
```

Comment:
This case does not catch FinitePowerPlant, and I think it is needed here, please correct me if I am wrong.

Reply:
[InfiniteBus](https://github.com/gridsingularity/d3a/pull/586/files#diff-82afe03c9b73bbef99c1ff6b646c24ddR52) is being separately dealt.

Reply:
InfiniteBus i a different strategy than FinitePowerPlant and MarketMakerStrategy though. The difference is that infinite bus is also a consumer, while the 2 other strategies are only producers.

Reply:
Sorry, mis-read in first place. You are right adding `FinitePowerPlant` too